src/Frontend/Core.hs

```diff
@@ -221,52 +235,113 @@ percentVotes idea numVoters vv = {- assert c -} v
           else (numVotes idea vv * 100) `div` numVoters
 
 
--- * render context
-
--- | Contains all the information which is needed to render a user role dependent functionality.
-data RenderContext = RenderContext
-      { _renderContextUser     :: User
-      }
-  deriving (Eq, Read, Show)
-
--- | Calculates the render context for role sensitive page rendering
-renderContext :: (ActionPersist m, ActionUserHandler m) => m RenderContext
-renderContext = RenderContext <$> currentUser
-
-
 -- * pages
 
+data AccessResult
+  = AccessGranted
+  | AccessDenied { _accessDeniedMsg :: ST, _accessDeniedRedirect :: Maybe URL }
```

Comment:
if you have a reason for the current implementation that i missed, please don't change anything.  otherwise the names you suggest are fine.